domain/src/main/java/org/oppia/android/domain/state/StateDeck.kt

```diff
@@ -128,14 +128,53 @@ internal class StateDeck internal constructor(
       .build()
     currentDialogInteractions.clear()
     pendingTopState = state
+    // Re-initialize the variable revealedHintIndex with -1 to indicate that no hint has been
+    // revealed on the new pendingTopState.
+    revealedHintIndex = -1
+    solutionIsRevealed = false
+  }
+
+  fun pushStateForHint(state: State, hintIndex: Int): EphemeralState {
+    val interactionBuilder = state.interaction.toBuilder().setHint(
+      hintIndex,
+      hintList.get(0)
+    )
+    val newState = state.toBuilder().setInteraction(interactionBuilder).build()
+    val ephemeralState = EphemeralState.newBuilder()
+      .setState(newState)
+      .setHasPreviousState(!isCurrentStateInitial())
+      .setPendingState(
+        PendingState.newBuilder().addAllWrongAnswer(currentDialogInteractions).addAllHint(hintList)
+      )
+      .build()
+    pendingTopState = newState
+    hintList.clear()
+    // Increment the value of revealHintIndex by 1 every-time a new hint is revealed.
+    revealedHintIndex++
+    return ephemeralState
+  }
+
+  fun pushStateForSolution(state: State): EphemeralState {
+    val interactionBuilder = state.interaction.toBuilder().setSolution(solution)
+    val newState = state.toBuilder().setInteraction(interactionBuilder).build()
+    val ephemeralState = EphemeralState.newBuilder()
+      .setState(newState)
+      .setHasPreviousState(!isCurrentStateInitial())
+      .setPendingState(
+        PendingState.newBuilder().addAllWrongAnswer(currentDialogInteractions).addAllHint(hintList)
+      )
+      .build()
+    pendingTopState = newState
+    solutionIsRevealed = true
+    return ephemeralState
```

Comment:
Did these come from a merge conflict? I don't think we want to reintroduce this code.

In general, please make sure to double check your changes before pushing to ensure exactly the changes you intend to submit are pushed.

Ditto elsewhere in this & the other changed files.

Reply:
I think you've fixed it in the latest change. All I see are the removals of 'internal' which seems correct.